src/metalperformanceshaders.cs

```diff
@@ -4203,159 +6066,2016 @@ interface MPSImageReduceUnary {
+	[BaseType (typeof (MPSNNGradientState), Name = "MPSCNNConvolutionGradientState")]
+	[DisableDefaultCtor]
+	interface MPSCnnConvolutionGradientState : MPSImageSizeEncodingState {
+
+		[Export ("gradientForWeights")]
+		IMTLBuffer GradientForWeights { get; }
+
+		[Export ("gradientForBiases")]
+		IMTLBuffer GradientForBiases { get; }
+
+		[Export ("convolution", ArgumentSemantic.Retain)]
+		MPSCnnConvolution Convolution { get; }
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSState), Name = "MPSCNNConvolutionWeightsAndBiasesState")]
+	[DisableDefaultCtor]
+	interface MPSCnnConvolutionWeightsAndBiasesState {
+
+		[Export ("weights")]
+		IMTLBuffer Weights { get; }
+
+		[NullAllowed, Export ("biases")]
+		IMTLBuffer Biases { get; }
+
+		[Export ("initWithWeights:biases:")]
+		IntPtr Constructor (IMTLBuffer weights, [NullAllowed] IMTLBuffer biases);
+
+		[Export ("initWithDevice:cnnConvolutionDescriptor:")]
+		IntPtr Constructor (IMTLDevice device, MPSCnnConvolutionDescriptor descriptor);
+
+		[Static]
+		[Export ("temporaryCNNConvolutionWeightsAndBiasesStateWithCommandBuffer:cnnConvolutionDescriptor:")]
+		MPSCnnConvolutionWeightsAndBiasesState GetTemporaryCNNConvolutionWeightsAndBiasesState (IMTLCommandBuffer commandBuffer, MPSCnnConvolutionDescriptor descriptor);
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSCnnGradientKernel), Name = "MPSCNNConvolutionGradient")]
+	[DisableDefaultCtor]
+	interface MPSCnnConvolutionGradient {
+
+		[Export ("sourceGradientFeatureChannels")]
+		nuint SourceGradientFeatureChannels { get; }
+
+		[Export ("sourceImageFeatureChannels")]
+		nuint SourceImageFeatureChannels { get; }
+
+		[Export ("groups")]
+		nuint Groups { get; }
+
+		[Export ("channelMultiplier")]
+		nuint ChannelMultiplier { get; }
+
+		[Export ("dataSource", ArgumentSemantic.Retain)]
+		IMPSCnnConvolutionDataSource DataSource { get; }
+
+		[Export ("gradientOption", ArgumentSemantic.Assign)]
+		MPSCnnConvolutionGradientOption GradientOption { get; set; }
+
+		[Deprecated (PlatformName.TvOS, 12, 0, message: "This doesn't do anything. It is here for backward compatibility. 'MPSCnnConvolutionGradient doesn't serialize weights. It gets weight from 'State.Convolution.DataSource' on first use i.e. first 'Encode' call.")]
+		[Deprecated (PlatformName.iOS, 12, 0, message: "This doesn't do anything. It is here for backward compatibility. 'MPSCnnConvolutionGradient doesn't serialize weights. It gets weight from 'State.Convolution.DataSource' on first use i.e. first 'Encode' call.")]
+		[Deprecated (PlatformName.MacOSX, 10, 14, message: "This doesn't do anything. It is here for backward compatibility. 'MPSCnnConvolutionGradient doesn't serialize weights. It gets weight from 'State.Convolution.DataSource' on first use i.e. first 'Encode' call.")]
+		[Export ("serializeWeightsAndBiases")]
+		bool SerializeWeightsAndBiases { get; set; }
```

Comment:
same - even more in this case since it seems it's not useful (and we don't have backward compatibility to consider since it's new bindings)

Reply:
Agreed, super short lived 11.3 -> 12.0